hathitrust/lss_solr_configs
# Dev 937 solrcloud embedded zookeeper

Sorry, this is a large PR... I will try to create small PRs in the future.

**The PR consists on setting up Solr8 server following different approaches. An independent directory have been created to cluster each architecture:**
* standalone mode ==> solr_standaole_mode
* [better for development] cloud model with an embedded zookeeper server ==> solrCloud_embedded_zooKeeper
* [better for production] claud model with an external zookeeper server ==> solrCloud_external_zooKeeper

In the README file you can find a detail explanation of each experiment

**How to test this PR?**

In the main folder:

- **Solr in standalone mode**
  ```
    docker build -t solr-text-search-8-standalone solr_standalone_mode
    docker-compose -f docker-compose_solr8_standalone_mode.yml up```

- **Solr embedded ZooKeeper**

    ```
      docker build -t full-text-search-embedded_zoo solrCloud_embedded_zooKeeper
       docker compose -f docker-compose_embedded_zooKeeper.yml up```

- **Solr external ZooKeeper**
    ```docker compose -f docker-compose_external_zooKeeper.yml up
         cd solrCloud_external_zooKeeper
        docker exec solr1 /var/solr/data/collection_manager.sh 
    ```

there should not find error in the logs

you should be able to access to this [link](http://localhost:8983/solr/#/core-x/query?q=*:*&q.op=OR&indent=true)
In the page you should see data, 1978 documents in total. See the attached image

<img width="805" alt="image" src="https://github.com/hathitrust/lss_solr_configs/assets/47088146/888565f1-b3fb-4382-a35f-aeac1374a51f">


If you want to test one of these solutions on babel-local-dev in the README file you can find the steps to integrate it.
